api/client/beacon/health.go

```diff
@@ -0,0 +1,42 @@
+package beacon
+
+import (
+	"sync"
+)
+
+type NodeHealth struct {
+	isHealthy     bool
+	HealthCh      chan bool
+	isHealthyLock sync.RWMutex
+}
+
+func NewNodeHealthTracker() *NodeHealth {
+	return &NodeHealth{
+		isHealthy: true, // just default it to true
```

Comment:
Defaulting to `false` is safer. Otherwise immediately after creating the struct things may start running as if the node is healthy, even if it's not. Better to wait for the node to explicitly signal that it's healthy.

Reply:
I think there's a 12 second buffer, i think it's better user experience for node runners to assume it's working first one ( which is what's happening right now anyways)

Reply:
We can check health at the start of `runHealthCheckRoutine` to get the initial state

Reply:
hmm maybe I need to add something for initializing the health tracker then. will think about it and add something if needed.